### ISSUE TITLE
Failed to send zip file

### ISSUE THREAD

**Issue by lemtea8** [localsend/localsend#788]
### Describe the bug

Similar to #567, file with zip suffix cannot be sent from desktop to phone.  
If rename the file extension from file.zip to file.jpg, then it can completed without error.  

The message from desktop client:  
```
flutter: 2023-10-24 14:29:43.947363 [WARNING] [Send] Error while sending file file.zip
flutter: DioException [bad response]: The request returned an invalid status code of 500.
flutter: #0      DioMixin.fetch.<anonymous closure> (package:dio/src/dio_mixin.dart:507)
#1      _RootZone.runUnary (dart:async/zone.dart:1661)
#2      _FutureListener.handleError (dart:async/future_impl.dart:174)
#3      Future._propagateToListeners.handleError (dart:async/future_impl.dart:852)
#4      Future._propagateToListeners (dart:async/future_impl.dart:873)
#5      Future._completeError (dart:async/future_impl.dart:649)
#6      _SyncCompleter._completeError (dart:async/future_impl.dart:60)
#7      _Completer.completeError (dart:async/future_impl.dart:26)
#8      Future.any.onError (dart:async/future.dart:620)
#9      _RootZone.runBinary (dart:async/zone.dart:1666)
#10     _FutureListener.handleError (dart:async/future_impl.dart:171)
#11     Future._propagateToListeners.handleError (dart:async/future_impl.dart:852)
#12     Future._propagateToListeners (dart:async/future_impl.dart:873)
#13     Future._completeError (dart:async/future_impl.dart:649)
#14     Future._chainForeignFuture.<anonymous closure> (dart:async/future_impl.dart:550)
#15     _microtaskLoop (dart:async/schedule_microtask.dart:40)
#16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49)
```

And the error from adb logcat:  
```
10-24 14:29:44.351  2834 29528 E MediaProvider: insertFileIfNecessary failed
10-24 14:29:44.351  2834 29528 E MediaProvider: java.lang.IllegalArgumentException: MIME type application/zip cannot be inserted into content://media/external_primary/video/media; expected MIME type under video/*
10-24 14:29:44.351  2834 29528 E MediaProvider:         at com.android.providers.media.MediaProvider.ensureFileColumns(MediaProvider.java:3824)
10-24 14:29:44.351  2834 29528 E MediaProvider:         at com.android.providers.media.MediaProvider.ensureUniqueFileColumns(MediaProvider.java:3595)
10-24 14:29:44.351  2834 29528 E MediaProvider:         at com.android.providers.media.MediaProvider.insertFile(MediaProvider.java:4367)
10-24 14:29:44.351  2834 29528 E MediaProvider:         at com.android.providers.media.MediaProvider.insertInternal(MediaProvider.java:4929)
10-24 14:29:44.351  2834 29528 E MediaProvider:         at com.android.providers.media.MediaProvider.insert(MediaProvider.java:4653)
10-24 14:29:44.351  2834 29528 E MediaProvider:         at com.android.providers.media.MediaProvider.insertFileForFuse(MediaProvider.java:9209)
10-24 14:29:44.351  2834 29528 E MediaProvider:         at com.android.providers.media.MediaProvider.insertFileIfNecessaryForFuse(MediaProvider.java:9304)
```

### To reproduce

1. Select a zip file from desktop
3. Send to phone
4. Set the receive directory to /sdcard/DCIM
5. Error occurs: `PathAccessException: Cannot open file, path = '/storage/emulated/0/DCIM/file.zip' (OS Error: Operation not permitted, errno = 1)`

### Expected behavior

Success without error.

### Desktop (please complete the following information)

- OS: Fedora 38, AppImage
- Version: 1.11.1(35)

### Smartphone (please complete the following information)

- OS: android 13
- Version: 1.11.1(35)

**Comment by prasadsunny1** [comment]
Android 13 does not give you full access to storage by default. Maintainers can probably look into it.

**Comment by bersace** [comment]
> Android 13 does not give you full access to storage by default. Maintainers can probably look into it.

l also encounter "Operation not permitted". But I cant find on which file/destination it fails. I'm wondering if it's about space or diacrictics. Here is a log of localsend installed from flatpak, sending to Android 13.

``` console
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:10.097495    [INFO] [HTTP]
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:10.098174    [INFO] [Send] Sending 16 - Introït: Dicit Dominum.ogg
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:10.098967    [INFO] [HTTP] *** Request ***
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:10.099144    [INFO] [HTTP] uri: https://192.168.1.84:53317/api/localsend/v2/upload?sessionId=759a6efa-bf40-4567-8f70-4a88d5461a02&fileId=d7a9fff6-7fd7-46df-befc-98d828fc57dd&token=0db61473-c548-4247-a859-75c03769deae
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:10.099254    [INFO] [HTTP] data:
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:10.099375    [INFO] [HTTP] Instance of '_FileStream'
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:10.099468    [INFO] [HTTP]
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: #0      DBusClient._callMethod (package:dbus/src/dbus_client.dart:1102)
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: <asynchronous suspension>
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: #1      DBusClient.callMethod (package:dbus/src/dbus_client.dart:608)
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: <asynchronous suspension>
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: #2      WakelockPlusLinuxPlugin.toggle (package:wakelock_plus/src/wakelock_plus_linux_plugin.dart:40)
nov. 12 18:43:10 gilwell org.localsend.localsend_app.desktop[5950]: <asynchronous suspension>
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.131868    [INFO] [HTTP] *** DioException ***:
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.132068    [INFO] [HTTP] uri: https://192.168.1.84:53317/api/localsend/v2/upload?sessionId=759a6efa-bf40-4567-8f70-4a88d5461a02&fileId=d7a9fff6-7fd7-46df-befc-98d828fc57dd&token=0db61473-c548-4247-a859-75c03769deae
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.132159    [INFO] [HTTP] DioException [bad response]: The request returned an invalid status code of 500.
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.132255    [INFO] [HTTP] uri: https://192.168.1.84:53317/api/localsend/v2/upload?sessionId=759a6efa-bf40-4567-8f70-4a88d5461a02&fileId=d7a9fff6-7fd7-46df-befc-98d828fc57dd&token=0db61473-c548-4247-a859-75c03769deae
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.132324    [INFO] [HTTP] Response Text:
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.132409    [INFO] [HTTP] {"message":"Could not save file"}
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.132484    [INFO] [HTTP]
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.132545    [INFO] [HTTP]
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.132788 [WARNING] [Send] Error while sending file 16 - Introït: Dicit Dominum.ogg
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: DioException [bad response]: The request returned an invalid status code of 500.
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: #0      DioMixin.fetch.<anonymous closure> (package:dio/src/dio_mixin.dart:507)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #1      _RootZone.runUnary (dart:async/zone.dart:1661)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #2      _FutureListener.handleError (dart:async/future_impl.dart:174)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #3      Future._propagateToListeners.handleError (dart:async/future_impl.dart:852)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #4      Future._propagateToListeners (dart:async/future_impl.dart:873)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #5      Future._completeError (dart:async/future_impl.dart:649)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #6      _SyncCompleter._completeError (dart:async/future_impl.dart:60)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #7      _Completer.completeError (dart:async/future_impl.dart:26)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #8      Future.any.onError (dart:async/future.dart:620)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #9      _RootZone.runBinary (dart:async/zone.dart:1666)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #10     _FutureListener.handleError (dart:async/future_impl.dart:171)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #11     Future._propagateToListeners.handleError (dart:async/future_impl.dart:852)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #12     Future._propagateToListeners (dart:async/future_impl.dart:873)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #13     Future._completeError (dart:async/future_impl.dart:649)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #14     Future._chainForeignFuture.<anonymous closure> (dart:async/future_impl.dart:550)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #15     _microtaskLoop (dart:async/schedule_microtask.dart:40)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: #16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49)
nov. 12 18:43:11 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:11.133240    [INFO] [Send] Transfer finished with errors.
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:24.821235    [INFO] [Server] Server already running.
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: flutter: 2023-11-12 18:43:24.821591    [INFO] [Multicast] Already listening to multicast
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: DioException [bad response]: The request returned an invalid status code of 403.
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #0      DioMixin.fetch.<anonymous closure> (package:dio/src/dio_mixin.dart:507)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #1      _RootZone.runUnary (dart:async/zone.dart:1661)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #2      _FutureListener.handleError (dart:async/future_impl.dart:174)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #3      Future._propagateToListeners.handleError (dart:async/future_impl.dart:852)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #4      Future._propagateToListeners (dart:async/future_impl.dart:873)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #5      Future._completeError (dart:async/future_impl.dart:649)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #6      _SyncCompleter._completeError (dart:async/future_impl.dart:60)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #7      _Completer.completeError (dart:async/future_impl.dart:26)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #8      Future.any.onError (dart:async/future.dart:620)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #9      _RootZone.runBinary (dart:async/zone.dart:1666)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #10     _FutureListener.handleError (dart:async/future_impl.dart:171)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #11     Future._propagateToListeners.handleError (dart:async/future_impl.dart:852)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #12     Future._propagateToListeners (dart:async/future_impl.dart:873)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #13     Future._completeError (dart:async/future_impl.dart:649)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #14     Future._chainForeignFuture.<anonymous closure> (dart:async/future_impl.dart:550)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #15     _microtaskLoop (dart:async/schedule_microtask.dart:40)
nov. 12 18:43:24 gilwell org.localsend.localsend_app.desktop[5950]: #16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49)
n
```

Hope this helps.

Regards,

**Comment by bersace** [comment]
Removing diacritics and punctuation did not help.

I also added privileges:

``` console
$ adb shell appops get org.localsend.localsend_app
Uid mode: READ_EXTERNAL_STORAGE: ignore
WRITE_EXTERNAL_STORAGE: ignore
LEGACY_STORAGE: ignore
WRITE_EXTERNAL_STORAGE: allow
READ_MEDIA_AUDIO: allow; time=+26m18s581ms ago
WRITE_MEDIA_AUDIO: deny; rejectTime=+26m19s343ms ago
READ_MEDIA_VIDEO: allow; time=+26m18s580ms ago
WRITE_MEDIA_VIDEO: deny; rejectTime=+26m19s341ms ago
READ_MEDIA_IMAGES: allow; time=+26m18s578ms ago
WRITE_MEDIA_IMAGES: deny; rejectTime=+25m0s339ms ago
LEGACY_STORAGE: allow
READ_DEVICE_IDENTIFIERS: deny; rejectTime=+2m13s54ms ago
ACCESS_MEDIA_LOCATION: allow; time=+26m18s576ms ago
MANAGE_EXTERNAL_STORAGE: default; rejectTime=+25m0s343ms ago
NO_ISOLATED_STORAGE: deny; rejectTime=+28m44s458ms ago
USE_ICC_AUTH_WITH_DEVICE_IDENTIFIER: default; rejectTime=+2m13s55ms ago
ACCESS_RESTRICTED_SETTINGS: allow; time=+32m27s824ms ago
```

**Comment by Tienisto** [comment]
Please checkout #710

**Comment by Tienisto** [comment]
This should be fixed in v1.13.0
